File: C++/Medium/WrongAnswer/find-all-lonely-numbers-in-the-array_4.cpp
class Solution {
public:
    vector<int> findLonely(vector<int>& nums) {
        
        vector<int> res;
        for(int i = 0; i < nums.size(); ++i) {
            if(i == 0) {
                if(nums[i] != nums[i+1]) {
                    res.push_back(nums[i]);
                }
            } else if(i == nums.size()-1) {
                if(nums[i] != nums[i-1]) {
                    res.push_back(nums[i]);
                }
            } else {
                if(nums[i] != nums[i-1] && nums[i] != nums[i+1]) {
                    res.push_back(nums[i]);
                }
            }
        }
        return res;
    }
};

File: Java/Medium/WrongAnswer/find-all-lonely-numbers-in-the-array_4.java
class Solution {
    public List<Integer> findLonely(int[] nums) {
        
        List<Integer> result = new ArrayList<>();
        if(nums.length == 1) {
            result.add(nums[0]);
            return result;
        }
        if(nums[0] != nums[1]) {
            result.add(nums[0]);
        }
        for(int i = 1; i < nums.length - 1; i++) {
            if(nums[i] != nums[i - 1] && nums[i] != nums[i + 1]) {
                result.add(nums[i]);
            }
        }
        if(nums[nums.length - 1] != nums[nums.length - 2]) {
            result.add(nums[nums.length - 1]);
        }
        return result;
    }
}